tracex/extraction/logic/orchestrator.py

```diff
@@ -102,21 +102,31 @@ def initialize_modules(self):
         print("Initialization of modules successful.")
         return modules
 
-    def run(self):
+    def run(self, view=None):
         """Run the modules."""
         modules = self.initialize_modules()
+        modules_number = len(modules) + 2
+        current_step = 0
 
         patient_journey = self.configuration.patient_journey
         if "preprocessing" in self.configuration.modules:
             preprocessor = self.configuration.modules.get("preprocessing")()
-            patient_journey = preprocessor.execute(patient_journey=self.configuration.patient_journey)
-
+            self.update_progress(view, current_step, modules_number, "Preprocessing")
+            patient_journey = preprocessor.execute(
+                patient_journey=self.configuration.patient_journey
+            )
+            current_step += 1
 
+        self.update_progress(view, current_step, modules_number, "Cohort Tagger")
         self.db_objects["cohort"] = self.configuration.modules[
             "cohort_tagging"
         ]().execute_and_save(self.data, patient_journey)
+        current_step += 1
         for module in modules:
+            self.update_progress(view, current_step, modules_number, module.name)
```

Comment:
The maximum number of modules is required once but passed for each execution of a module. Although we a only talking about a number here, it is generally discouraged to pass arguments multiple times if they don't change and are only required one time. Try to fix this.

Reply:
The only way is to use a class variable but I'm don't think that this would be the right use case.

Reply:
I change module_number to be an instance variable. But in retrospective its doesnt change anything at all. The passing arguments doesnt reduce and instead of passing a number we are passing a whole instance.

Reply:
If we use the modules list from the configuration instead (like i suggested in the other thread we don`t need that variable at all. we can access `len(self.configuration.modules)` from within the `update_progress()` function.

If we use the instance variable, it must be defined in the `__init__` function. Something like `self.modules_number: int = 0`.

Reply:
@FR-SON
I'm not sure if this solves the "problem". To access len(self.configuration.modules) we have to pass "self" to the function. So in the end we passing the whole instance everytime instead of one number, I guess

Reply:
I think the only solution is to create a class variable if we really want to reduce the amount of passing arguments (but not sure if thats the right use case)

Reply:
@FR-SON thx for the suggestion, in the end I think it doesnt solves the problem to reduce the amount of passing parameters. Instead of passing a number we are passing the whole instance, so its basically just an indirection. 

But this solution is definitely better and already implemented.